### PR TITLE
feat: IPNet type

### PR DESCRIPTION
**Pull request by kindermoumoute**

### CODE REVIEW

File: scw/custom_types.go [hunk 1]
```diff
@@ -164,3 +166,36 @@ func (tsp *TimeSeriesPoint) UnmarshalJSON(b []byte) error {
 
 	return nil
 }
+
+// IPNet inherits net.IPNet and represents an IP network.
+type IPNet struct {
```
Comment (jerome-quere) [inline scw/custom_types.go]: should it be a type alias ?

Comment (kindermoumoute) [reply scw/custom_types.go]: This way the type inherits `net.IPNet` methods

File: scw/custom_types.go [hunk 2]
```diff
@@ -164,3 +166,36 @@ func (tsp *TimeSeriesPoint) UnmarshalJSON(b []byte) error {
...
+}
+
+func (n *IPNet) UnmarshalJSON(b []byte) error {
+	var str *string
+
+	err := json.Unmarshal(b, &str)
+	if err != nil {
+		return err
+	}
+	if str == nil {
```
Comment (jerome-quere) [inline scw/custom_types.go]: I am not sure this can be happen

File: scw/custom_types.go [hunk 3]
```diff
@@ -164,3 +166,36 @@ func (tsp *TimeSeriesPoint) UnmarshalJSON(b []byte) error {
...
+		return nil
+	}
+
+	if !strings.Contains(*str, "/") && net.ParseIP(*str) != nil {
+		*str += "/32"
```
Comment (jerome-quere) [inline scw/custom_types.go]: do not work with ipv6

File: scw/custom_types.go (hunk 2 above)
Comment (kindermoumoute) [reply scw/custom_types.go]: `{"ip_range":null}`

**Pull request merged**